tests/testcore/dynamic_config_overrides.go

```diff
@@ -0,0 +1,83 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package testcore
+
+import (
+	"time"
+
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/persistence/visibility"
+)
+
+var (
+	// Functional tests don't use any dynamic config files. All settings get their default values
+	// (defined where setting is declared), besides those which are overridden.
+	// There are 4 ways to override a setting:
+	// 1. Globally using this file. Every test suite creates a new test cluster using this overrides.
+	// 2. Per test suite using SetupSuiteWithCluster() and WithDynamicConfigOverrides() option.
+	// 3. Per test using s.OverrideDynamicConfig() method.
+	// 4. Per specific cluster per test (if test has more than one default cluster) using cluster.OverrideDynamicConfig() method.
+	//
+	// NOTE: settings which are not really dynamic (requires server restart to take effect) can't be overridden on test level.
```

Comment:
Main idea of this PR: describe how to override DC and consolidate all global overrides in one file with clear name (for easy discovery).